meshnet/server.go

```diff
@@ -1340,22 +1339,37 @@ func (s *Server) RenamePeer(
 		}, nil
 	}
 
-	if req.Nickname == "" && peer.Nickname == "" {
-		log.Println(internal.InfoPrefix, "peer doesn't have a nickname")
+	if req.Nickname == "" {
+		if peer.Nickname == "" {
+			log.Println(internal.InfoPrefix, "peer doesn't have already a nickname")
```

Comment:
Is this remove nickname when it is not set situation? Should be say to the user about that instead pretending that operation succeeded?

Reply:
changed and will return an error in this case